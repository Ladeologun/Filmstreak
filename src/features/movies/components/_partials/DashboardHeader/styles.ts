import { StyleSheet } from "react-native";
import { COLORS, Fonts } from "~styles";


const styles = StyleSheet.create({
    container:{
        height:50,
        flexDirection:"row",
        backgroundColor:COLORS.PRIMARY,
        alignItems:"center",
        justifyContent:"center",
        paddingHorizontal:15
    },
    image:{
        height:"100%",
        width:50,
        margin:0,
    },
    text:{
        fontSize:22,
        color:COLORS.WHITE,
        fontWeight:"700",
        letterSpacing:1,
        fontFamily:Fonts.GupterBold,
        
    },
   
    textcontainer:{
        flex:1,
        flexDirection:"row",
        justifyContent:"center",
        alignItems:"center"
        // backgroundColor:"red",
        // paddingRight:30
    }
})

export default styles;