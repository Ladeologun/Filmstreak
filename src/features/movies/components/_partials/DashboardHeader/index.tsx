import React from 'react';
import { View, Text } from 'react-native';
import styles from './styles';
import { COLORS } from '~styles';

interface Iprops{
    title:string;
    isLoading?:boolean
}

const DashboardHeader: React.FC<Iprops> =({title,isLoading}) =>{

    return (
        <View style={styles.container}>
            <View style={styles.textcontainer}>
                <Text style={[styles.text,{color:COLORS.SKYBLUE}, isLoading && {opacity:0.3}]}>{title}</Text>
            </View>
        </View>
    );
}


export default DashboardHeader;