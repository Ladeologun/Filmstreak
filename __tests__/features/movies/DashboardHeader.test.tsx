import renderer from 'react-test-renderer';
import {test,expect} from '@jest/globals';
import DashboardHeader from '~movies/components/_partials/DashboardHeader';


test('renders correctly', () => {
    const tree = renderer.create(<DashboardHeader title='filmstreak' />).toJSON();
    expect(tree).toMatchSnapshot();
});
