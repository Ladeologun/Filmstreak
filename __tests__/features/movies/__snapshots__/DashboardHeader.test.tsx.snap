// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#0F111D",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "paddingHorizontal": 15,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
            "fontFamily": "Gupter-Bold",
            "fontSize": 22,
            "fontWeight": "700",
            "letterSpacing": 1,
          },
          {
            "color": "skyblue",
          },
          undefined,
        ]
      }
    >
      filmstreak
    </Text>
  </View>
</View>
`;
